test/ts/utils/fs.test.ts

```diff
@@ -0,0 +1,189 @@
+import * as chai from 'chai';
+import * as fs from 'fs/promises';
+
+import {
+    copyFileWithDir,
+    normalizePath,
+    readJsonFile,
+    readPathsRecursively,
+    validateDir,
+    validateFile,
+} from '../../../dist/utils/fs';
+import mockFs = require('mock-fs');
+const { expect } = chai;
+
+describe('normalizePath', () => {
+    it('windows short path name format normalizes', () => {
+        const path = 'D:\\very short path';
+        const expected = 'D:/very short path';
+
+        expect(normalizePath(path)).to.equal(expected);
+    });
+
+    it('unix path name format normalizes (no change)', () => {
+        const path = '/home/unix/path';
+
+        expect(normalizePath(path)).to.equal(path);
+    });
+
+    it('windows long path name format should not normalize', () => {
+        const path = '\\\\?\\D:\\very long path';
+
+        expect(normalizePath(path)).to.equal(path);
+    });
+});
+
+describe('readJsonFile', () => {
+    afterEach(()=>{
+        chai.spy.restore();
+    });
+    
+    it('should read a valid JSON file', async () => {
+        const vcConfigContent = {
+            runtime: 'edge',
+            entrypoint: 'index.js',
+        };
+        chai.spy.on(fs,'readFile',(path,encoding)=>{
```

Comment:
we should omit this vars, maybe use `_`.

Reply:
Unused vars was removed